habitat_baselines/common/obs_transformers.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import abc
+import copy
+import numbers
+from typing import List
+
+import torch
+from gym.spaces.dict_space import Dict as SpaceDict
+from torch import nn
+
+from habitat.config import Config
+from habitat.core.logging import logger
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.common.utils import (
+    center_crop,
+    get_image_height_width,
+    image_resize_shortest_edge,
+    overwrite_gym_box_shape,
+)
+
+
+class ObservationTransformer(nn.Module, metaclass=abc.ABCMeta):
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+
+    def transform_observation_space(
+        self, observation_space: SpaceDict, **kwargs
+    ):
+        return observation_space
+
+    @classmethod
+    @abc.abstractmethod
+    def from_config(cls, config, envs):
+        pass
+
+    def forward(self, *args):
+        return args
+
+
+@baseline_registry.register_obs_transformer()
+class ResizeShortestEdge(ObservationTransformer):
+    def __init__(self, size: int, channels_last: bool = False):
+        r"""An nn module the resizes your the shortest edge of the input while maintaining aspect ratio.
+        This module assumes that all images in the batch are of the same size.
+        Args:
+            size: The size you want to resize the shortest edge to
+            channels_list: indicates if channels is the last dimension
+        """
+        super().__init__()
+        self._size = size
+        self.channels_last = channels_last
+
+    def transform_observation_space(
+        self,
+        observation_space: SpaceDict,
+        trans_keys=("rgb", "depth", "semantic"),
```

Comment:
Makes more sense to have list here.
```suggestion
        trans_keys: List[str]=["rgb", "depth", "semantic"],
```

Reply:
Should be a tuple per flake8 not a list.